### PR TITLE
DAOS-4589: src/include/daos/pool.h: remove the client-side dc_pool_evict() functionality

### PR DESCRIPTION
**Pull request by cibervicho**
In favor of using 'dmg pool evict', the dc_pool_evict() method has been removed
from the client-side.

In 'pool/evict_test.py', the test_evict_bad_server_name(self) test has been skipped.
This is because the scenario where trying to evict a pool with an invalid server name,
 such as --sys=BAD_SERVER_NAME is not working correctly.

### CODE REVIEW

**Comment by cibervicho** [comment]
**Failing test**: Test / Functional_Hardware_Large / MacsioTest.4-./io/macsio_test.py:MacsioTest.test_macsio_daos_vol;container-dfuse-hosts-job_manager_timeout-macsio-openmpi-pool-servers-fb7e
**Reported in**: https://jira.hpdd.intel.com/browse/DAOS-6138

File: src/tests/ftest/pool/evict_test.py
Comment (berserk-fury) [inline src/tests/ftest/pool/evict_test.py:113]: perhaps, dated comment?
It seems that we are now using the dmg command rather than the API

File: src/tests/ftest/pool/evict_test.py
Comment (berserk-fury) [inline src/tests/ftest/pool/evict_test.py:90]: Could you create a story to re-enable it and then link it to DAOS-5545?

**Comment by GitHuaKuang** [comment]
> **Failing test**: Test / Functional_Hardware_Large / MacsioTest.4-./io/macsio_test.py:MacsioTest.test_macsio_daos_vol;container-dfuse-hosts-job_manager_timeout-macsio-openmpi-pool-servers-fb7e
> **Reported in**: https://jira.hpdd.intel.com/browse/DAOS-6138

I noticed there are multiple backtraces similar to the failure in my PR-4983(not exactly the same), somehow I couldn't find this bt in daos-6138:
07:29:25 DEBUG| [stderr] [1,5]<stderr>:12/11-07:29:25.02 wolf-117 DAOS[25780/25780] client EMRG src/common/tse.c:280 tse_sched_fini() Assertion 'dsp->dsp_inflight == 0' failed
07:29:25 DEBUG| [stderr] [1,5]<stderr>:macsio: src/common/tse.c:280: tse_sched_fini: Assertion `dsp->dsp_inflight == 0' failed.
07:29:25 DEBUG| [stderr] [1,5]<stderr>:[wolf-117:25780] *** Process received signal ***
07:29:25 DEBUG| [stderr] [1,5]<stderr>:[wolf-117:25780] Signal: Aborted (6)
07:29:25 DEBUG| [stderr] [1,5]<stderr>:[wolf-117:25780] Signal code:  (-6)
07:29:25 DEBUG| [stderr] [1,5]<stderr>:[wolf-117:25780] [ 0] /lib64/libc.so.6(+0x36400)[0x7f89c3c2e400]
07:29:25 DEBUG| [stderr] [1,5]<stderr>:[wolf-117:25780] [ 1] /lib64/libc.so.6(gsignal+0x37)[0x7f89c3c2e387]
07:29:25 DEBUG| [stderr] [1,5]<stderr>:[wolf-117:25780] [ 2] /lib64/libc.so.6(abort+0x148)[0x7f89c3c2fa78]
07:29:25 DEBUG| [stderr] [1,5]<stderr>:[wolf-117:25780] [ 3] /lib64/libc.so.6(+0x2f1a6)[0x7f89c3c271a6]
07:29:25 DEBUG| [stderr] [1,5]<stderr>:[wolf-117:25780] [ 4] /lib64/libc.so.6(+0x2f252)[0x7f89c3c27252]
07:29:25 DEBUG| [stderr] [1,5]<stderr>:[wolf-117:25780] [ 5] /lib64/libdaos_common.so(tse_sched_fini+0x346)[0x7f89962d1686]
07:29:25 DEBUG| [stderr] [1,5]<stderr>:[wolf-117:25780] [ 6] /usr/lib64/openmpi3/lib/libhdf5_vol_daos.so.1.0.0(+0x619fd)[0x7f89968949fd]
07:29:25 DEBUG| [stderr] [1,5]<stderr>:[wolf-117:25780] [ 7] /usr/lib64/openmpi3/lib/libhdf5_vol_daos.so.1.0.0(+0x66297)[0x7f8996899297]
07:29:25 DEBUG| [stderr] [1,5]<stderr>:[wolf-117:25780] [ 8] /usr/lib64/openmpi3/lib/libhdf5.so.201(+0x40b9bd)[0x7f89c4bf09bd]
07:29:25 DEBUG| [stderr] [1,5]<stderr>:[wolf-117:25780] [ 9] /usr/lib64/openmpi3/lib/libhdf5.so.201(H5VL_file_create+0x9e)[0x7f89c4bfbaee]
07:29:25 DEBUG| [stderr] [1,5]<stderr>:[wolf-117:25780] [10] /usr/lib64/openmpi3/lib/libhdf5.so.201(H5Fcreate+0x26a)[0x7f89c4946b0a]
07:29:25 DEBUG| [stderr] [1,5]<stderr>:[wolf-117:25780] [11] /usr/lib64/openmpi3/bin/macsio[0x41c207]
07:29:25 DEBUG| [stderr] [1,5]<stderr>:[wolf-117:25780] [12] /usr/lib64/openmpi3/bin/macsio[0x41a92f]
07:29:25 DEBUG| [stderr] [1,5]<stderr>:[wolf-117:25780] [13] /usr/lib64/openmpi3/bin/macsio(main+0x82f)[0x40d4af]
07:29:25 DEBUG| [stderr] [1,5]<stderr>:[wolf-117:25780] [14] /lib64/libc.so.6(__libc_start_main+0xf5)[0x7f89c3c1a555]
07:29:25 DEBUG| [stderr] [1,5]<stderr>:[wolf-117:25780] [15] /usr/lib64/openmpi3/bin/macsio[0x40dc60]
07:29:25 DEBUG| [stderr] [1,5]<stderr>:[wolf-117:25780] *** End of error message ***

File: src/tests/ftest/pool/evict_test.py
Comment (cibervicho) [reply src/tests/ftest/pool/evict_test.py]: Done. Thank you for catching this.

File: src/tests/ftest/pool/evict_test.py
Comment (cibervicho) [reply src/tests/ftest/pool/evict_test.py:90]: @berserk-fury Sure, I was planning to do it once this PR is approved. Do you want me to create it before this?

**Review by berserk-fury — APPROVED** [review]

File: src/client/api/init.c
Comment (GitHuaKuang) [inline src/client/api/init.c:56]: I made some changes in this area after discussion with the team in https://github.com/daos-stack/daos/pull/3984, you may want to take a look, I think it applies here as weel.

Comment (cibervicho) [reply src/client/api/init.c:56]: Thank you for sharing @GitHuaKuang, working on this.

**Comment by cibervicho** [comment]
Based on review from @GitHuaKuang and his work in #3984 I made some changes. Waiting for CI checks to pass.

**Comment by cibervicho** [comment]
- Build phase passed successfully.
- Unit Test phase completed.
- Functional Test in CentOS and Leap finished successfully so far:

Functional on CentOS 7
======================
`[2020-12-14T22:42:21.172Z] Running: avocado run --ignore-missing-references on --html-job-result on --tap-job-result off --filter-by-tags=pr,-hw --mux-yaml /tmp/tmpJ7x73x/pool-evict_test.yaml -- ./pool/evict_test.py
[2020-12-14T22:42:21.433Z] JOB ID     : 01c5a7abc2ece5869700784193521c3ed3ae161a
[2020-12-14T22:42:21.433Z] JOB LOG    : /var/tmp/ftest/avocado/job-results/job-2020-12-14T22.42-01c5a7a/job.log
[2020-12-14T22:42:48.867Z]  (1/2) ./pool/evict_test.py:EvictTests.test_evict;badparams-container-hosts-pool-server_config-56da:  PASS (27.20 s)
[2020-12-14T22:43:11.172Z]  (2/2) ./pool/evict_test.py:EvictTests.test_evict_bad_uuid;badparams-container-hosts-pool-server_config-56da:  PASS (19.69 s)
[2020-12-14T22:43:11.172Z] RESULTS    : PASS 2 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
[2020-12-14T22:43:11.172Z] JOB TIME   : 47.38 s`

Functional on Leap 15
=====================
`[2020-12-14T22:36:25.019Z] Running: avocado run --ignore-missing-references on --html-job-result on --tap-job-result off --filter-by-tags=pr,-hw --mux-yaml /tmp/tmpRat4_e/pool-evict_test.yaml -- ./pool/evict_test.py
[2020-12-14T22:36:25.390Z] JOB ID     : 20dd307483f522814f355ce34f3d03d2f62d674b
[2020-12-14T22:36:25.390Z] JOB LOG    : /var/tmp/ftest/avocado/job-results/job-2020-12-14T22.36-20dd307/job.log
[2020-12-14T22:36:57.822Z]  (1/2) ./pool/evict_test.py:EvictTests.test_evict;badparams-container-hosts-pool-server_config-120c:  PASS (29.02 s)
[2020-12-14T22:37:16.512Z]  (2/2) ./pool/evict_test.py:EvictTests.test_evict_bad_uuid;badparams-container-hosts-pool-server_config-120c:  PASS (21.29 s)
[2020-12-14T22:37:16.512Z] RESULTS    : PASS 2 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
[2020-12-14T22:37:16.512Z] JOB TIME   : 50.91 s`

File: src/include/daos_task.h
Comment (jolivier23) [inline src/include/daos_task.h:52]: I think rather than adding dummy entries, we can just do something like...

DAOS_OPC_SET_PARAMS = /* expected value */, and skip the old entry

File: src/pool/cli.c
Comment (jolivier23) [inline src/pool/cli.c:1530]: I wonder if we need to invoke task_complete here ?

Either way, in @GitHuaKuang 's patch, I asked him to name it dc_deprecated and maybe print a warning or error.

**Review by jolivier23 — CHANGES_REQUESTED** [review]

File: src/pool/cli.c
Comment (GitHuaKuang) [reply src/pool/cli.c:1532]: We should be able to share the same dc_deprecated() in my PR-3984, no need for each individual functions.

Comment (cibervicho) [reply src/pool/cli.c:1532]: @GitHuaKuang, I agree, I need to wait for your changes to be landed or merge your changes to my PR.

File: src/tests/ftest/pool/evict_test.py
Comment (phender) [inline src/tests/ftest/pool/evict_test.py:285]: If it can be avoided, we should not be commenting out tests, but rather skipping them with `skipForTicket()` or `cancelForTicket()`.  When we comment out tests we lose the ability to track that they are not being run.  Since in this case we want to skip running an entire test method, the `skipForTicket()` should be used:
```
    @skipForTicket("DAOS-5545")
    def test_evict_bad_server_name(self):
```

File: src/tests/ftest/pool/evict_test.py
Comment (phender) [inline src/tests/ftest/pool/evict_test.py:90]: In general, commenting out test code to avoid temporarily running a test should be avoided.

In this case however I don't think commenting out is needed at all as these commented out sections of `def evict_badparam(self, test_param)` will never get executed.  This method is called [here](https://github.com/daos-stack/daos/blob/be7081af201ae233ff926bff2eb31e24e794e345/src/tests/ftest/pool/evict_test.py#L285) and [here](https://github.com/daos-stack/daos/blob/be7081af201ae233ff926bff2eb31e24e794e345/src/tests/ftest/pool/evict_test.py#L295).  The first call should be handled by adding `@skipForTicket("DAOS-5545")` to the `def test_evict_bad_server_name(self)` method.  In the second call, the passed in `test_param` will only ever be [`"invalid_uuid"`](https://github.com/daos-stack/daos/blob/master/src/tests/ftest/pool/evict_test.yaml#L23).

**Review by phender — CHANGES_REQUESTED** [review]

File: src/tests/ftest/pool/evict_test.py
Comment (cibervicho) [reply src/tests/ftest/pool/evict_test.py:90]: That is true Phil. Working on this.

File: src/tests/ftest/pool/evict_test.py
Comment (cibervicho) [reply src/tests/ftest/pool/evict_test.py:285]: Thank you.

File: src/tests/ftest/pool/evict_test.py
Comment (cibervicho) [reply src/tests/ftest/pool/evict_test.py]: @berserk-fury I created the ticket DAOS-6356 requesting this.
https://jira.hpdd.intel.com/browse/DAOS-6356
Thank you!

**Comment by cibervicho** [comment]
Functional Tests, in both "CentoOS 7" and "Leap 15", finished successfully:

`[2020-12-16T19:40:58.257Z] Running: avocado run --ignore-missing-references on --html-job-result on --tap-job-result off --filter-by-tags=pr,-hw --mux-yaml /tmp/tmpkDf9Lh/pool-evict_test.yaml -- ./pool/evict_test.py
[2020-12-16T19:40:58.516Z] JOB ID     : 1f44bb5d436c13c481aeadc0dfb913b7dde1d7e7
[2020-12-16T19:40:58.516Z] JOB LOG    : /var/tmp/ftest/avocado/job-results/job-2020-12-16T19.40-1f44bb5/job.log
[2020-12-16T19:41:30.637Z]  (1/3) ./pool/evict_test.py:EvictTests.test_evict;badparams-container-hosts-pool-server_config-cc2a:  PASS (28.01 s)
[2020-12-16T19:41:30.637Z]  (2/3) ./pool/evict_test.py:EvictTests.test_evict_bad_server_name;badparams-container-hosts-pool-server_config-cc2a:  SKIP
[2020-12-16T19:41:48.740Z]  (3/3) ./pool/evict_test.py:EvictTests.test_evict_bad_uuid;badparams-container-hosts-pool-server_config-cc2a:  PASS (21.75 s)
[2020-12-16T19:41:48.740Z] RESULTS    : PASS 2 | ERROR 0 | FAIL 0 | SKIP 1 | WARN 0 | INTERRUPT 0 | CANCEL 0
[2020-12-16T19:41:48.740Z] JOB TIME   : 50.43 s
[2020-12-16T19:41:48.740Z] JOB HTML   : /var/tmp/ftest/avocado/job-results/job-2020-12-16T19.40-1f44bb5/results.html
[2020-12-16T19:41:48.740Z] Total test time: 50s

[2020-12-16T19:39:30.891Z] Running: avocado run --ignore-missing-references on --html-job-result on --tap-job-result off --filter-by-tags=pr,-hw --mux-yaml /tmp/tmpdlVS4F/pool-evict_test.yaml -- ./pool/evict_test.py
[2020-12-16T19:39:31.154Z] JOB ID     : 746e70d79895928447222d425947f4f2481daaa0
[2020-12-16T19:39:31.154Z] JOB LOG    : /var/tmp/ftest/avocado/job-results/job-2020-12-16T19.39-746e70d/job.log
[2020-12-16T19:40:03.573Z]  (1/3) ./pool/evict_test.py:EvictTests.test_evict;badparams-container-hosts-pool-server_config-a033:  PASS (29.54 s)
[2020-12-16T19:40:03.573Z]  (2/3) ./pool/evict_test.py:EvictTests.test_evict_bad_server_name;badparams-container-hosts-pool-server_config-a033:  SKIP
[2020-12-16T19:40:21.679Z]  (3/3) ./pool/evict_test.py:EvictTests.test_evict_bad_uuid;badparams-container-hosts-pool-server_config-a033:  PASS (20.31 s)
[2020-12-16T19:40:21.680Z] RESULTS    : PASS 2 | ERROR 0 | FAIL 0 | SKIP 1 | WARN 0 | INTERRUPT 0 | CANCEL 0
[2020-12-16T19:40:21.680Z] JOB TIME   : 50.64 s
[2020-12-16T19:40:21.941Z] JOB HTML   : /var/tmp/ftest/avocado/job-results/job-2020-12-16T19.39-746e70d/results.html
[2020-12-16T19:40:21.941Z] Total test time: 51s`

File: src/tests/ftest/pool/evict_test.py
Comment (berserk-fury) [inline src/tests/ftest/pool/evict_test.py:117]: We are mixing DAOS error codes and errno codes. Is it expected?

Comment (cibervicho) [reply src/tests/ftest/pool/evict_test.py:117]: I didn't have the opportunity to test the 'BAD_SERVER_NAME' scenario, only the bad UUID, that is why I only modified the this expected error.
I'm currently checking the dmg code under "/src/control/cmd/dmg/" to see if there are other return codes managed by the DMG command.

Comment (cibervicho) [reply src/tests/ftest/pool/evict_test.py:117]: While checking this with Jonathan (@berserk-fury) found that the function "exitWithError" in the /src/control/cmd/dmg/main.go L186 DMG always returns 1.
@mjmac is this correct? Are we missing some other place where different error codes are returned by DMG?

Comment (mjmac) [reply src/tests/ftest/pool/evict_test.py:117]: dmg only returns 0 (success) or 1 (error). If you need to get at the DAOS error code you need to use the `-j,--json` flag and look at the status field. This is how the dmg helper on the C side works: https://github.com/daos-stack/daos/blob/master/src/common/tests_dmg_helpers.c#L233

Comment (cibervicho) [reply src/tests/ftest/pool/evict_test.py:117]: Thank you!

Comment (cibervicho) [reply src/tests/ftest/pool/evict_test.py:117]: Since we are moving to 'dmg' and this command responds back with rc=0 or rc=1, I'm updating the evict_test.py to be consistent with that behavior and see if command succeeds or fails with correct or bad parameters.

Comment (cibervicho) [reply src/tests/ftest/pool/evict_test.py]: @berserk-fury, this are the results in Functional after the changes:

`[2020-12-17T06:48:02.366Z] Running: avocado run --ignore-missing-references on --html-job-result on --tap-job-result off --filter-by-tags=pr,-hw --mux-yaml /tmp/tmpo1HCzn/pool-evict_test.yaml -- ./pool/evict_test.py
[2020-12-17T06:48:02.624Z] JOB ID     : 727ae6ad2d0dc8326f90cc34cd0af1166534b821
[2020-12-17T06:48:02.624Z] JOB LOG    : /var/tmp/ftest/avocado/job-results/job-2020-12-17T06.48-727ae6a/job.log
[2020-12-17T06:48:34.706Z]  (1/3) ./pool/evict_test.py:EvictTests.test_evict;badparams-container-hosts-pool-server_config-120c:  PASS (31.43 s)
[2020-12-17T06:48:34.706Z]  (2/3) ./pool/evict_test.py:EvictTests.test_evict_bad_server_name;badparams-container-hosts-pool-server_config-120c:  SKIP
[2020-12-17T06:48:56.635Z]  (3/3) ./pool/evict_test.py:EvictTests.test_evict_bad_uuid;badparams-container-hosts-pool-server_config-120c:  PASS (21.00 s)
[2020-12-17T06:48:56.636Z] RESULTS    : PASS 2 | ERROR 0 | FAIL 0 | SKIP 1 | WARN 0 | INTERRUPT 0 | CANCEL 0
[2020-12-17T06:48:56.636Z] JOB TIME   : 53.09 s
[2020-12-17T06:48:56.636Z] JOB HTML   : /var/tmp/ftest/avocado/job-results/job-2020-12-17T06.48-727ae6a/results.html
[2020-12-17T06:48:56.636Z] Total test time: 53s`

**Review by berserk-fury — DISMISSED** [review]

**Review by berserk-fury — DISMISSED** [review]

**Review by jolivier23 — APPROVED** [review]
Needs a merge due to conflict with Hua's patch

**Comment by cibervicho** [comment]
> Needs a merge due to conflict with Hua's patch

Working on that. Thank you Jeff

**Review by berserk-fury — APPROVED** [review]

**Review by GitHuaKuang — APPROVED** [review]

**Review by saurabhtandan — APPROVED** [review]
Test code LGTM

**Comment by cibervicho** [comment]
Dear @daos-stack/daos-gatekeeper, the functional tests for CentOS 7 and Leap 15 finished successfully and the evict_test, involved in this PR passed as expected:

`[2020-12-17T23:07:43.818Z] Running: avocado run --ignore-missing-references on --html-job-result on --tap-job-result off --filter-by-tags=pr,-hw --mux-yaml /tmp/tmpPjPGha/pool-evict_test.yaml -- ./pool/evict_test.py
[2020-12-17T23:07:43.818Z] JOB ID     : 17d08264d0e289f1813ef11b29d8a9a8baca0785
[2020-12-17T23:07:43.818Z] JOB LOG    : /var/tmp/ftest/avocado/job-results/job-2020-12-17T23.07-17d0826/job.log
[2020-12-17T23:08:16.210Z]  (1/3) ./pool/evict_test.py:EvictTests.test_evict;badparams-container-hosts-pool-server_config-a033:  PASS (28.17 s)
[2020-12-17T23:08:16.210Z]  (2/3) ./pool/evict_test.py:EvictTests.test_evict_bad_server_name;badparams-container-hosts-pool-server_config-a033:  SKIP
[2020-12-17T23:08:34.357Z]  (3/3) ./pool/evict_test.py:EvictTests.test_evict_bad_uuid;badparams-container-hosts-pool-server_config-a033:  PASS (20.09 s)
[2020-12-17T23:08:34.357Z] RESULTS    : PASS 2 | ERROR 0 | FAIL 0 | SKIP 1 | WARN 0 | INTERRUPT 0 | CANCEL 0
[2020-12-17T23:08:34.357Z] JOB TIME   : 48.90 s
[2020-12-17T23:08:34.357Z] JOB HTML   : /var/tmp/ftest/avocado/job-results/job-2020-12-17T23.07-17d0826/results.html
[2020-12-17T23:08:34.357Z] Total test time: 49s`

`[2020-12-17T23:02:00.686Z] Running: avocado run --ignore-missing-references on --html-job-result on --tap-job-result off --filter-by-tags=pr,-hw --mux-yaml /tmp/tmpqDRbNA/pool-evict_test.yaml -- ./pool/evict_test.py
[2020-12-17T23:02:00.686Z] JOB ID     : 977bde7f70b6385df057c5c3ff7be8ff077be197
[2020-12-17T23:02:00.686Z] JOB LOG    : /var/tmp/ftest/avocado/job-results/job-2020-12-17T23.01-977bde7/job.log
[2020-12-17T23:02:32.778Z]  (1/3) ./pool/evict_test.py:EvictTests.test_evict;badparams-container-hosts-pool-server_config-120c:  PASS (30.50 s)
[2020-12-17T23:02:32.778Z]  (2/3) ./pool/evict_test.py:EvictTests.test_evict_bad_server_name;badparams-container-hosts-pool-server_config-120c:  SKIP
[2020-12-17T23:02:54.720Z]  (3/3) ./pool/evict_test.py:EvictTests.test_evict_bad_uuid;badparams-container-hosts-pool-server_config-120c:  PASS (21.62 s)
[2020-12-17T23:02:54.720Z] RESULTS    : PASS 2 | ERROR 0 | FAIL 0 | SKIP 1 | WARN 0 | INTERRUPT 0 | CANCEL 0
[2020-12-17T23:02:54.720Z] JOB TIME   : 52.94 s
[2020-12-17T23:02:54.720Z] JOB HTML   : /var/tmp/ftest/avocado/job-results/job-2020-12-17T23.01-977bde7/results.html
[2020-12-17T23:02:54.720Z] Total test time: 54s`

**Review by jolivier23 — APPROVED** [review]

**Review by jolivier23 — APPROVED** [review]
Please add gatekeeper back when testing is complete and ready to land.   Otherwise, it just clutters our list

**Comment by cibervicho** [comment]
All Functional Hardware Medium tests passed and some others were skipped.
The error reported seems related to launch.py L1426, which is not related to this commit:

`[2020-12-19T00:39:39.695Z]   Errors detected running "set -eu; rc=0; copied=(); for file in /var/tmp/core.*; do if [ -e $file ]; then if [ ! -s $file ]; then; ((rc++)); ls -al $file; else if sudo chmod 644 $file && scp $file wolf-52:/var/tmp/ftest/avocado/job-results/latest/stacktraces/${file##*/}-$(hostname -s); then copied+=($file); if ! sudo rm -fr $file; then ((rc++)); ls -al $file; fi; else ((rc++)); ls -al $file; fi; fi; fi; done; echo Copied ${copied[@]:-no files}; exit $rc":
[2020-12-19T00:39:39.695Z] There are 2 lines of output
[2020-12-19T00:39:39.695Z]     wolf-[52,122-125]: rc=1, output:
[2020-12-19T00:39:39.695Z]       bash: -c: line 0: syntax error near unexpected token `;'
[2020-12-19T00:39:39.695Z]       bash: -c: line 0: `set -eu; rc=0; copied=(); for file in /var/tmp/core.*; do if [ -e $file ]; then if [ ! -s $file ]; then; ((rc++)); ls -al $file; else if sudo chmod 644 $file && scp $file wolf-52:/var/tmp/ftest/avocado/job-results/latest/stacktraces/${file##*/}-$(hostname -s); then copied+=($file); if ! sudo rm -fr $file; then ((rc++)); ls -al $file; fi; else ((rc++)); ls -al $file; fi; fi; fi; done; echo Copied ${copied[@]:-no files}; exit $rc'
[2020-12-19T00:39:39.695Z] ERROR: Detected one or more tests that failed archiving!`

**Comment by cibervicho** [comment]
> All Functional Hardware Medium tests passed and some others were skipped.
> The error reported seems related to launch.py L1426, which is not related to this commit:
> 
> `[2020-12-19T00:39:39.695Z] Errors detected running "set -eu; rc=0; copied=(); for file in /var/tmp/core.*; do if [ -e $file ]; then if [ ! -s $file ]; then; ((rc++)); ls -al $file; else if sudo chmod 644 $file && scp $file wolf-52:/var/tmp/ftest/avocado/job-results/latest/stacktraces/${file##*/}-$(hostname -s); then copied+=($file); if ! sudo rm -fr $file; then ((rc++)); ls -al $file; fi; else ((rc++)); ls -al $file; fi; fi; fi; done; echo Copied ${copied[@]:-no files}; exit $rc": [2020-12-19T00:39:39.695Z] There are 2 lines of output [2020-12-19T00:39:39.695Z] wolf-[52,122-125]: rc=1, output: [2020-12-19T00:39:39.695Z] bash: -c: line 0: syntax error near unexpected token `;'
> [2020-12-19T00:39:39.695Z] bash: -c: line 0: `set -eu; rc=0; copied=(); for file in /var/tmp/core.*; do if [ -e $file ]; then if [ ! -s $file ]; then; ((rc++)); ls -al $file; else if sudo chmod 644 $file && scp $file wolf-52:/var/tmp/ftest/avocado/job-results/latest/stacktraces/${file##*/}-$(hostname -s); then copied+=($file); if ! sudo rm -fr $file; then ((rc++)); ls -al $file; fi; else ((rc++)); ls -al $file; fi; fi; fi; done; echo Copied ${copied[@]:-no files}; exit $rc' [2020-12-19T00:39:39.695Z] ERROR: Detected one or more tests that failed archiving!`

This error got fixed in  https://github.com/daos-stack/daos/pull/4141 40 minutes ago.

**Comment by cibervicho** [comment]
Dear @daos-stack/daos-gatekeeper this PR finished the Test stage with no new errors, but the Functional HW Medium one which was an error fixed and merged into master a couple of minutes ago.

**Comment by cibervicho** [comment]
Hi @daos-stack/daos-gatekeeper, could you help me landing this PR? It was required for version 1.2 Community Release and I'm not sure if there is going to be a code freeze soon.
This PR finished the test stage with no issues. The only one was Functional HW Medium, which is not related to the changes I made here, and it was fixed in https://github.com/daos-stack/daos/pull/4141. Thank you.

**Comment by cibervicho** [comment]
Dear @daos-stack/daos-gatekeeper, can you help me landing this PR please? It has 4 approvals and the test phase has finished. In my comments above I mentioned the only error found was taken care in #4141 which at this point is already merged. Thank you again.

**Comment by ashleypittman** [comment]
> Dear @daos-stack/daos-gatekeeper, can you help me landing this PR please? It has 4 approvals and the test phase has finished. In my comments above I mentioned the only error found was taken care in #4141 which at this point is already merged. Thank you again.

Builds should be stable now, and progressing much faster than before, so could you merge this with master and it should have up-to-date results later today.  Performing a merge with master does not cause the PR to lose the approvals either.

**Comment by cibervicho** [comment]
> > Dear @daos-stack/daos-gatekeeper, can you help me landing this PR please? It has 4 approvals and the test phase has finished. In my comments above I mentioned the only error found was taken care in #4141 which at this point is already merged. Thank you again.
> 
> Builds should be stable now, and progressing much faster than before, so could you merge this with master and it should have up-to-date results later today. Performing a merge with master does not cause the PR to lose the approvals either.

Got it!. Working on that.

**Comment by cibervicho** [comment]
Dear @daos-stack/daos-gatekeeper, I merged with master and re-pushed to CI. Now CI checks finished successfully. Could you help me landing this PR please?
Thank you in advance!

**Pull request merged**
